### PR TITLE
feat: config v3 & exit on batch input completion

### PR DESCRIPTION
**Pull request by Ryan-Git**
see #32 
gravity-operator should be adjusted with the new config api after this is released.

### CODE REVIEW

File: cmd/gravity/single_binary.go [hunk 1]
```diff
@@ -123,6 +118,14 @@ func main() {
 		syscall.SIGTERM,
 		syscall.SIGQUIT)
 
+	if cfg.PipelineConfig.InputPlugin.Mode == config.Batch {
+		go func(server *gravity.Server) {
+			<-server.Input.Done()
+			server.Close()
+			os.Exit(0)
```
Comment (ming535) [inline cmd/gravity/single_binary.go:125]: gravity-operator is not aware of this right now, what will happen?

File: gravity/config/config_v3.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package config
+
+import "encoding/json"
+
+type PipelineConfigV3 struct {
```
Comment (ming535) [inline gravity/config/config_v3.go:7]: Suggest add a `version` for the configuration schema version, just like docker compose file version: https://docs.docker.com/compose/compose-file/compose-versioning/

File: gravity/config/config_v2.go [hunk 3]
```diff
@@ -13,6 +13,51 @@ type PipelineConfigV2 struct {
 	SchedulerPlugins map[string]interface{} `mapstructure:"scheduler" toml:"scheduler" json:"scheduler,omitempty"`
 }
 
+func (c *PipelineConfigV2) IsV3() bool {
```
Comment (ming535) [inline gravity/config/config_v2.go:17]: Use `log.Warn` when the configuration version is not `v3`.

File: gravity/config/config_v3.go (hunk 2 above)
Comment (Ryan-Git) [reply gravity/config/config_v3.go:7]: added

File: gravity/config/config_v3.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package config
+
+import "encoding/json"
+
+const PipelineConfigV3Version = "1"
```
Comment (ming535) [inline gravity/config/config_v3.go]: Can we use "1.0" as the string?

Comment (Ryan-Git) [reply gravity/config/config_v3.go]: ok

**Review by ming535 — APPROVED** [review]

**Pull request merged**